pkg/mkverification-raw-efi/spec-verification.sh

```diff
@@ -0,0 +1,547 @@
+    label="${label#*: }"
+    cat <<__EOT__
+    {
+      "ztype": 255,
+      "phylabel": "NVME${ID}",
+      "assigngrp": "${grp}",
+      "phyaddrs": {
+        "PciLong": "${NVME}"
+      },
+      "logicallabel": "${label}",
+      "usagePolicy": {}
+__EOT__
+    if [ -n "$verbose" ]; then
+        add_pci_info "${NVME}"
+    fi
+    cat <<__EOT__
+    },
+__EOT__
+    ID=$(( ${ID:-0} + 1 ))
+done
+
+#enumerate serial ports
+ID="1"
+for TTY in /sys/class/tty/*; do
+   if [ -f "$TTY/device/resources" ]; then
+      IO=$(grep '^io ' "$TTY/device/resources" | sed -e 's#io 0x##' -e 's#0x##')
+      IRQ=$(awk '/^irq /{print $2;}' < "$TTY/device/resources")
+   elif [ "$(uname -m)" = aarch64 ] && [ -f "$TTY/irq" ]; then
+      IRQ=$(cat "$TTY/irq")
+      [ "${IRQ:-0}" -gt 0 ] || IRQ=""
+      IO=""
+   else
+      IO=""
+      IRQ=""
+   fi
+   TTY=$(echo "$TTY" | cut -f5 -d/)
+   if [ -n "$IO" ] || [ -n "$IRQ" ]; then
+cat <<__EOT__
+    {
+      "ztype": 3,
+      "phylabel": "COM${ID}",
+      "assigngrp": "COM${ID}",
+      "phyaddrs": {
+__EOT__
+      if [ -n "$IO" ] && [ -n "$IRQ" ]; then
+cat <<__EOT__
+        "Ioports": "${IO}",
+        "Irq": "${IRQ}",
+__EOT__
+      fi
+cat <<__EOT__
+        "Serial": "/dev/${TTY}"
+      },
+      "logicallabel": "COM${ID}",
+      "usagePolicy": {}
+    },
+__EOT__
+     ID=$(( ${ID:-0} + 1 ))
+   fi
+done
+
+#enumerate cellular modems
+ID="0"
+for USBDEV in /sys/class/usbmisc/*; do
+    get_modem_protocol "$USBDEV" >/dev/null || continue
+    IFNAME="$(get_modem_ifname "$USBDEV")"
+    MODEMS="${MODEMS}${IFNAME}\n" # skip during NIC enumeration
+    USB_ADDR="$(get_modem_usbaddr "$USBDEV")"
+    PCI_ADDR="$(get_modem_pciaddr "$USBDEV")"
+    GROUP=$(get_assignmentgroup "${IFNAME}" "${PCI_ADDR}")
+    label=$(lspci -D | grep ${PCI_ADDR})
+    label="${label#*: }"
+    cat <<__EOT__
+    ${COMMA}
+    {
+      "ztype": 6,
+      "logicallabel": "${label}",
+      "phylabel": "modem${ID}",
+      "assigngrp": "${GROUP}",
+      "phyaddrs": {
+        "PciLong": "${PCI_ADDR}",
+        "UsbAddr": "${USB_ADDR}"
+      },
+      "cost": 10,
+__EOT__
+    if [ -n "$verbose" ] && [ -x "$(command -v picocom)" ]; then
+        AT_PORT="$(get_modem_atport "$USB_ADDR")"
+        if [ -n "$AT_PORT" ]; then
+            # Close any previous communication at this AT port.
+            send_at_command "+++" "${AT_PORT}"
+            # Return modem identification.
+            # Use sed to remove lines containing only whitespace.
+            OUTPUT="$(send_at_command "ATI" "${AT_PORT}" | sed '/^\s*$/d')"
+            if echo "$OUTPUT" | grep -qv "ERROR"; then
+                # Remove control commands.
+                OUTPUT="$(echo "$OUTPUT" | sed '/^\s*ATI\s*$/d; /^\s*OK\s*$/d' | tr -d '\r')"
+                # Keep at most two lines and join them by semi-colon.
+                OUTPUT="$(echo "$OUTPUT" | head -n 2 | sed ':a;N;$!ba;s/\n/; /g')"
+                cat <<__EOT__
+      "description": "$OUTPUT",
```

Comment:
If you want to describe peripherals for the use, add `description` field like I did here.